ethers-signers/src/trezor/app.rs

```diff
@@ -56,9 +61,41 @@ impl TrezorEthereum {
         Ok(())
     }
 
+    fn get_cached_session() -> Result<Option<Vec<u8>>, TrezorError> {
+        let mut session = [0; 32];
+
+        let path = env::current_dir()
+            .map_err(|e| TrezorError::CacheError(e.to_string()))?
+            .join("cache")
+            .join("trezor.session");
+
+        if let Ok(file) = fs::File::open(path) {
+            let mut file = std::io::BufReader::new(file);
+            file.read(&mut session).map_err(|e| TrezorError::CacheError(e.to_string()))?;
```

Comment:
don't think a BufReader is necessary here for 32bytes?
also read does not guarantee to fill the buf entirely.
there is https://doc.rust-lang.org/stable/std/io/trait.Read.html#method.read_exact instead, or alternatively File::read_to_end